test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -569,39 +569,108 @@ static public void testWithFieldOnNonExistentClass() throws Throwable {
 		} catch (NoClassDefFoundError e) {}
 	}
 
-	/*
-	 * TODO: behaviour of the test between two valueTypes will depend on the new spec(not finalized)
-	 * 
-	 * Test ifacmp on value class
-	 * 
-	 * class TestIfacmpOnValueClass {}
-	 *
-	 *
-	 *	@Test(priority=2)
-	 *	static public void TestIfacmpOnValueClass() throws Throwable {
-	 *	int x = 0;
-	 *	int y = 0;
-	 *
-	 *	Object valueType = makePoint2D.invoke(x, y);
-	 *	Object refType = (Object) x;
-	 *
-	 *	Assert.assertFalse((valueType == refType), "An identity (==) comparison that contains a valueType should always return false");
-	 *
-	 *	Assert.assertFalse((refType == valueType), "An identity (==) comparison that contains a valueType should always return false");
-	 *
-	 *	Assert.assertFalse((valueType == valueType), "An identity (==) comparison that contains a valueType should always return false");
-	 *
-	 *	Assert.assertTrue((refType == refType), "An identity (==) comparison on the same refType should always return true");
-	 *
-	 *	Assert.assertTrue((valueType != refType), "An identity (!=) comparison that contains a valueType should always return true");
-	 *
-	 *	Assert.assertTrue((refType != valueType), "An identity (!=) comparison that contains a valueType should always return true");
-	 *
-	 *	Assert.assertTrue((valueType != valueType), "An identity (!=) comparison that contains a valueType should always return true");
-	 *
-	 *	Assert.assertFalse((refType != refType), "An identity (!=) comparison on the same refType should always return false");
-	 *	}
-	 */
+	@Test(priority=5)
+	static public void TestIfacmp() throws Throwable {
```

Comment:
can you spilt this into multiple tests:
- basic test (no value types)
- fast value type tests (only primitives and array references)
- potentially recursive test (j.l.Object and interfaces fields, but no value type instances)
- potentially recursive test with value types (j.l.Object and interfaces fields with value type instances)
- recursive tests (value type fields)
- floating point tests (value types with float and double fields)

Reply:
make sure the floating point tests include, NaNs, +-Zero and infinities